src/v/serde/serde.h

```diff
@@ -319,6 +333,14 @@ void write(iobuf& out, T t) {
     } else if constexpr (is_pair_v<Type>) {
         write(out, std::move(t.first));
         write(out, std::move(t.second));
+    } else if constexpr (is_duration_v<Type>) {
+        // We explicitly serialize it as ns to avoid any surprises like
+        // seastar updating underlying duration types without
+        // notice. See https://github.com/redpanda-data/redpanda/pull/5002
+        auto d = std::chrono::duration_cast<std::chrono::nanoseconds>(t);
```

Comment:
Anything that has previously serialised a `std::chrono::milliseconds` will now read it back wrong.

Reply:
using nanoseconds here would be ideal, but ben is right about milliseconds being baked in. however, i'm wondering if we can use nanoseconds in serde because presumably there isn't any current serde users?

Reply:
> however, i'm wondering if we can use nanoseconds in serde because presumably there isn't any current serde users?

I initially overlooked the ms regression but this is a good point. If this is true, would be nice to preserve ns precision and that extends neatly to time_points too.